cmd/agentctl/main.go

```diff
@@ -454,7 +454,8 @@ config that may be used with this agent.`,
 
 			cli := grafanacloud.NewClient(nil, apiKey, apiURL)
 
-			ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+			// setting timeout 2x as the default HTTP transport timeout (30s)
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second*60)
```

Comment:
Sorry, I meant setting the read timeout on the HTTP client itself:

```go
cli := &http.Client{
  Timeout: time.Minute, 
}

cli := grafanacloud.NewClient(cli, apiKey, apiUrl) 

cfg, err := cli.AgentConfig(context.Background(), stackID)
// ...
```

Reply:
I think the timeout on the http.Client should give more detailed error messages; let me double check.

Reply:
OK, yeah, it looks like http.Client.Timeout will add `(Client.Timeout exceeded while awaiting headers)` to the error messages, while the context itself won't.

Reply:
Cool, thanks for checking!